### ISSUE TITLE
Generate embedding without GPU

### ISSUE THREAD

**Issue by FlamingoZh** [facebookresearch/swav#36]
Hi, I love the project a lot, thanks for sharing it!

However, for some reason, the GPU is not available to me. I have downloaded the pretrained model, and my ultimate goal is to get some visual embedding from the pretrained model,  so I just wondering if there is any easy way to do the following thing: If I input a single image into the model, the model would output a corresponding embedding. Note that I don't expect to re-train or fine-tune the model, so maybe it's still possible to do the job without using a GPU. 

Also, if this is not too dumb, could you please specify the input/output size? Thank you very much!

**Comment by mathildecaron31** [comment]
Hi @FlamingoZh 
Thanks for your interest in our project.

You can load the model weights on cpu with the following:
`model_state_dict = torch.load(path_to_pretrained_model, map_location="cpu")`

The input to the model should be a tensor of dimension [B, 3, H, W] where B is the batch size, H and W are the image resolution (typically H=W=224).
The output of the model will be a tuple of `embedding` and `output`:
https://github.com/facebookresearch/swav/blob/b4aa051475144700d9532a61d9a61654ea236c52/main_swav.py#L290

`embedding` is a normalized tensor of dimension [B, feat_dim] (usually feat_dim=128).
`output` is the dot-product between the embeddings and the prototypes, it is of dimension [B, nmb_prototypes] (usually nmb_prototypes=3000)

Hope that helps

**Comment by FlamingoZh** [comment]
That's so helpful, thank you very much!

Also, just for clarification, is the "Imagenet" dataset mentioned in the paper actually the Imagenet ILSVRC2012? Thanks!

**Comment by mathildecaron31** [comment]
Yes, we pre-train swav on the training set of the ImageNet ILSVRC-2012 dataset.

**Comment by escorciav** [comment]
hi @FlamingoZh!

Could you please share more details of your experiments? I'm interested on that 👇 
https://github.com/facebookresearch/swav/issues/68
